custom/ilsgateway/tasks.py

```diff
@@ -63,28 +63,39 @@ def get_locations(api_object, facilities):
 
 
 def sync_product_stock(domain, endpoint, facility, checkpoint, date, limit=100, offset=0):
+    """
+    Syncs ProductStock objects in ILSGateway to StockState objects in CommTrack
+    """
     has_next = True
     next_url = ""
-    while has_next:
-        supply_point = facility
-        case = SupplyPointCase.view('hqcase/by_domain_external_id',
-                                    key=[domain, str(supply_point)],
-                                    reduce=False,
-                                    include_docs=True,
-                                    limit=1).first()
-        meta, product_stocks = endpoint.get_productstocks(
-            next_url_params=next_url,
-            limit=limit,
-            offset=offset,
-            filters=dict(supply_point=supply_point, last_modified__gte=date)
-        )
-        save_stock_data_checkpoint(checkpoint,
-                                   'product_stock',
-                                   meta.get('limit') or limit,
-                                   meta.get('offset') or offset,
-                                   date, facility, True)
-        for product_stock in product_stocks:
-            if case:
+    supply_point = facility
+    case = SupplyPointCase.view('hqcase/by_domain_external_id',
+                                key=[domain, str(supply_point)],
+                                reduce=False,
+                                include_docs=True,
+                                limit=1).first()
+    if case:
+        while has_next:
+            meta, product_stocks = endpoint.get_productstocks(
+                next_url_params=next_url,
+                limit=limit,
+                offset=offset,
+                filters=dict(supply_point=supply_point, last_modified__gte=date)
+            )
+            # todo: kkrampa shouldn't we wait to save the checkpoint until after we've processed all the data?
+            # otherwise seems like we would miss things
+            save_stock_data_checkpoint(checkpoint,
+                                       'product_stock',
+                                       meta.get('limit') or limit,
+                                       meta.get('offset') or offset,
+                                       date, facility, True)
+            for product_stock in product_stocks:
+                # this logic updates the StockState object based on data from the current ProductStock
+                # todo: It seems a little bit odd/wrong that we would have this in the migration code at all
+                # Shouldn't we be able to rely on the stock transaction logic loading the right data
+                # into the StockState?
+
```

Comment:
I think you're right that this is unnecessary. We don't calculate consumption during stock transaction but after that we should be able to generate StockState data